workspaces/bulk-import/plugins/bulk-import-backend/src/service/handlers/import/bulkImports.ts

```diff
@@ -45,18 +50,22 @@ type FindAllImportsResponse =
   | Components.Schemas.Import[]
   | Components.Schemas.ImportJobListV2;
 
-function sortImports(imports: Components.Schemas.Import[]) {
+function sortImports(
+  imports: Components.Schemas.Import[],
+  sortColumn: string = 'repository.name',
+  sortOrder: string = SortingOrderEnum.ASC,
+) {
   imports.sort((a, b) => {
-    if (a.repository?.name === undefined && b.repository?.name === undefined) {
-      return 0;
-    }
-    if (a.repository?.name === undefined) {
-      return -1;
-    }
-    if (b.repository?.name === undefined) {
-      return 1;
-    }
-    return a.repository.name.localeCompare(b.repository.name);
+    const value1 = getNestedValue(a, sortColumn);
+    const value2 = getNestedValue(b, sortColumn);
+    // Handle cases where values are undefined
+    if (value1 === undefined && value2 === undefined) return 0;
+    if (value1 === undefined) return SortingOrderEnum.ASC ? -1 : 1;
+    if (value2 === undefined) return SortingOrderEnum.ASC ? 1 : -1;
+    // Compare values based on sort order
+    return SortingOrderEnum.ASC === sortOrder
+      ? value1.localeCompare(value2)
+      : value2.localeCompare(value1);
   });
 }
```

Comment:
We should eliminate this and add the sorting parameter to the GitHub / Catalog API calls.

Reply:
Github APIs not supporting sorting on these params ,